extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -148,12 +169,25 @@
 
                 // Set protocol
                 GuacamoleConfiguration config = new GuacamoleConfiguration();
+                
+                GuacamoleProxyConfiguration proxyConfig;
+                try {
+                    proxyConfig = new LocalEnvironment().getDefaultGuacamoleProxyConfiguration();
```

Comment:
The `LocalEnvironment` should not be repeatedly recreated if it can be avoided.

A singleton instance of it is actually already bound during startup of the LDAP extension, so that would be the instance of choice:

https://github.com/apache/guacamole-client/blob/78f1ae1b4eac25501d532ddee94fd1d8588e56dc/extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/LDAPAuthenticationProviderModule.java#L74

That singleton instance can be injected within this class if needed, for example:

https://github.com/apache/guacamole-client/blob/78f1ae1b4eac25501d532ddee94fd1d8588e56dc/extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ConfigurationService.java#L41-L45

Since this is configuration, though, I'd think it would make more sense to provide this via a function within `ConfigurationService`.

Reply:
I originally went this route, but thought it might be a little overkill :-).  I'll go back to this method...

Reply:
Okay, switched back to this method.